### PR TITLE
Extract pipeline resolving code into a reusable class

### PR DESCRIPTION
**Pull request by bernd**
- Introduce a reusable PipelineResolver class that doesn't depend on the database
- Make the rule, pipeline, and stage metric prefix configurable

/prd https://github.com/Graylog2/graylog-plugin-enterprise/pull/7525
/nocl Internal refactoring

### CODE REVIEW

**Review by danotorrey — APPROVED** [review]
This regression tests with Illuminate processing and with a standard simple pipeline test successfully. The code looks good from what I can see, but it would be good to get one more review.

**Review by thll — APPROVED** [review]
LGTM

File: graylog2-server/src/main/java/org/graylog/plugins/pipelineprocessor/processors/PipelineResolverConfig.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+import org.graylog.plugins.pipelineprocessor.ast.Rule;
+import org.graylog.plugins.pipelineprocessor.db.PipelineDao;
+import org.graylog.plugins.pipelineprocessor.db.RuleDao;
+import org.graylog.plugins.pipelineprocessor.rest.PipelineConnections;
+
+import java.util.function.Supplier;
+import java.util.stream.Stream;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Strings.isNullOrEmpty;
```
[the remark this answers is not in the document]
Comment (bernd) [reply graylog2-server/src/main/java/org/graylog/plugins/pipelineprocessor/processors/PipelineResolverConfig.java]: Good catch, thank you!

**Comment by bernd** [comment]
Thanks for the reviews! I am moving it to draft again to fix the issue with metrics registration and removal.

Both are done in different places now which is confusing.

**Comment by bernd** [comment]
@danotorrey @thll The PR is ready for review again. I extracted the metrics registration and removal into a separate class.

**Review by danotorrey — APPROVED** [review]
LGTM and tests successfully with the latest

**Review by thll — CHANGES_REQUESTED** [review]

File: graylog2-server/src/main/java/org/graylog/plugins/pipelineprocessor/processors/PipelineResolver.java [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+            stage.setPipeline(pipeline);
+            stage.registerMetrics(pipelineMetricRegistry, pipeline.id());
+        });
+
+        pipeline.registerMetrics(pipelineMetricRegistry);
+
+        return pipeline;
+    }
+
+    public record Result(MetricRegistry metrics, ImmutableMap<String, Pipeline> pipelines) {}
```
[the remark this answers is not in the document]
Comment (bernd) [reply graylog2-server/src/main/java/org/graylog/plugins/pipelineprocessor/processors/PipelineResolver.java]: Good catch! I removed it.

**Review by thll — APPROVED** [review]

**Pull request merged**